### PR TITLE
[stable/velero] fix velero installation on azure landscape - add miss…

### PR DESCRIPTION
**Pull request by ahmadhajali**
What this PR does / why we need it:
Installing velero on K8S using helm does not provide all the required secret and env values which causes pod failures due to missing env values.
time="2019-10-30T17:34:51Z" level=info msg="Checking that all backup storage locations are valid" logSource="pkg/cmd/server/server.go:412"
An error occurred: some backup storage locations are invalid: error getting backup store for location "default": rpc error: code = Unknown desc = unable to get all required environment variables: the following keys do not have values: AZURE_TENANT_ID, AZURE_CLIENT_ID, AZURE_CLIENT_SECRET, AZURE_SUBSCRIPTION_ID

Which issue this PR fixes
This PR fixes the following issue: vmware-tanzu/velero#2021

### CODE REVIEW

File: stable/velero/templates/deployment.yaml [hunk 1]
```diff
@@ -90,11 +90,13 @@ spec:
...
                   apiVersion: v1
                   fieldPath: metadata.namespace
-          {{- if and .Values.credentials.useSecret (or (eq $provider "aws") (eq $provider "gcp")) }}
+          {{- if and .Values.credentials.useSecret (or (eq $provider "aws") (eq $provider "gcp") (eq $provider "azure")) }}
             {{- if eq $provider "aws" }}
             - name: AWS_SHARED_CREDENTIALS_FILE
-            {{- else }}
+            {{- else if eq $provider "gcp" }}
             - name: GOOGLE_APPLICATION_CREDENTIALS
+            {{- else }}
```
Comment (hectorj2f) [inline stable/velero/templates/deployment.yaml]: please leave the else as it is and check for azure instead.

Comment (hectorj2f) [reply stable/velero/templates/deployment.yaml]: In other words, move the azure env variable inside of a specific if statement for this provider

Comment (ahmadhajali) [reply stable/velero/templates/deployment.yaml]: azure is now checked with else if and gcp kept in else as before

**Comment by ahmadhajali** [comment]
I did 2 signoffs for both commits, but dco-labeler check still complains that there is a missing signoff. How should I resolve this issue?

**Comment by ahmadhajali** [comment]
What else needed from my side?

**Comment by ahmadhajali** [comment]
Hi, Is there an estimation to when this pull request will be approved since it is a showstopper for our azure support?

**Comment by zanhsieh** [comment]
@ahmadhajali 
You probably need another version bump:
```
I1111 17:35:00.222] Old chart version: 2.3.1
I1111 17:35:00.222] New chart version: 2.3.1
I1111 17:35:00.222] Error linting and installing charts: Error processing charts
I1111 17:35:00.222] ------------------------------------------------------------------------------------------------------------------------
I1111 17:35:00.223]  ✖︎ velero => (version: "2.3.1", path: "stable/velero") > Chart version not ok. Needs a version bump!
I1111 17:35:00.223] ------------------------------------------------------------------------------------------------------------------------
```

**Comment by jaygridley** [comment]
Hello, @ahmadhajali Can you also look into Restic deployment? Particullary lines https://github.com/helm/charts/blob/master/stable/velero/templates/restic-daemonset.yaml#L38 and https://github.com/helm/charts/blob/master/stable/velero/templates/restic-daemonset.yaml#L73 should be also fixed.

I am not sure about https://github.com/helm/charts/blob/master/stable/velero/templates/deployment.yaml#L73. This one should be also treated.

**Comment by solomson** [comment]
Think I found a workaround tested on app version 1.2, chart version 2.6.0
Basically combines the two methods.

```
cat << EOF > cloud
AZURE_SUBSCRIPTION_ID=${AZ_SUBSCRIPTION_ID}
AZURE_TENANT_ID=${AZ_TENANT_ID}
AZURE_CLIENT_ID=${AZ_BACKUP_SP_ID}
AZURE_CLIENT_SECRET=${AZ_BACKUP_SP_PASSWORD}
AZURE_RESOURCE_GROUP=${AZ_CLUSTER_GROUP}
AZURE_CLOUD_NAME=AzurePublicCloud
EOF

kubectl create secret generic velero-credentials \
    --namespace velero \
    --from-literal AZURE_SUBSCRIPTION_ID=${AZ_SUBSCRIPTION_ID} \
    --from-literal AZURE_TENANT_ID=${AZ_TENANT_ID} \
    --from-literal AZURE_CLIENT_ID=${AZ_BACKUP_SP_ID} \
    --from-literal AZURE_CLIENT_SECRET=${AZ_BACKUP_SP_PASSWORD} \
    --from-literal AZURE_RESOURCE_GROUP=${AZ_CLUSTER_GROUP} > /dev/null

kubectl create secret generic velero-credentials --from-file=cloud -n velero --dry-run -o yaml | kubectl apply -f - > /dev/null || true
& rm -f cloud
```

image

I had the errors listed above earlier, and now they seem to be gone.

**Comment by davidkarlsen** [comment]
This chart has been deprecated and moved to a new repo: https://github.com/helm/charts/blob/master/stable/velero/README.md

**Pull request closed without merging**